packages/beacon-node/src/chain/historicalState/worker.ts

```diff
@@ -32,145 +24,21 @@ const logger = getNodeLogger(workerData.loggerOpts);
 logger.info("Historical state worker started");
 
 const config = createBeaconConfig(chainConfigFromJson(workerData.chainConfigJson), workerData.genesisValidatorsRoot);
-
+const diffLayers = new DiffLayers(workerData.diffLayers);
 const db = new BeaconDb(config, await LevelDbController.create({name: workerData.dbLocation}, {logger}));
 
 const abortController = new AbortController();
 
 // Set up metrics, nodejs, state transition, queue
 const metricsRegister = workerData.metricsEnabled ? new RegistryMetricCreator() : null;
-let historicalStateRegenMetrics: HistoricalStateRegenMetrics | undefined;
+const historicalStateRegenMetrics: HistoricalStateRegenMetrics | undefined = metricsRegister
+  ? getMetrics(metricsRegister)
+  : undefined;
 let queueMetrics: QueueMetrics | undefined;
 if (metricsRegister) {
   const closeMetrics = collectNodeJSMetrics(metricsRegister, "lodestar_historical_state_worker_");
   abortController.signal.addEventListener("abort", closeMetrics, {once: true});
 
-  historicalStateRegenMetrics = {
```

Comment:
make sure we have equivalent metrics with new approach if applicable

Reply:
yes added new metrics related to it. 

1. State Diff Size
2. State Snapshot Size